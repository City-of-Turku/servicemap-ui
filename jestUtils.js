import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });