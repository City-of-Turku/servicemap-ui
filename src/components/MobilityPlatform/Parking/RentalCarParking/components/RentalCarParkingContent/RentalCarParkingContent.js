import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';
import TextComponent from '../../../../TextComponent';

const RentalCarParkingContent = ({ classes, intl, item }) => {
  const renderText = (msgId, value) => (
    <div className={classes.margin}>
      <Typography variant="body2">
        {value ? intl.formatMessage({ id: msgId }, { value }) : intl.formatMessage({ id: msgId })}
      </Typography>
    </div>
  );

  const formatName = name => name?.split(',')[0];

  const renderAccessInfo = (accessValue) => {
    const accessValueLower = accessValue.toLowerCase();
    if (accessValueLower === 'vapaa pääsy') {
      return renderText('mobilityPlatform.content.publicParking.access');
    }
    return null;
  };

  const translations = {
    placesTotal: 'mobilityPlatform.content.disabledParking.amount',
    totalTime: 'mobilityPlatform.content.publicParking.totalTime',
  };

  const names = {
    fi: formatName(item.name_fi),
    en: formatName(item.name_en),
    sv: formatName(item.name_sv),
  };

  const addressObj = {
    fi: item.address_fi,
    en: item.address_en,
    sv: item.address_sv,
  };

  return (
    <div className={classes.container}>
      <div className={classes.headerContainer}>
        <TextComponent textObj={names} isTitle />
      </div>
      <div className={classes.textContainer}>
        <TextComponent messageId="mobilityPlatform.content.address" textObj={addressObj} />
        {renderText(translations.placesTotal, item.extra.Paikkoja_y)}
        {item.extra.Max_aika_h ? renderText(translations.totalTime, item.extra.Max_aika_h) : null}
        {item.extra.Rajoit_lis ? <TextComponent textObj={item.extra.Rajoit_lis} /> : null}
        {renderAccessInfo(item.extra.Saavutetta)}
      </div>
    </div>
  );
};

RentalCarParkingContent.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  item: PropTypes.objectOf(PropTypes.any),
};

RentalCarParkingContent.defaultProps = {
  item: {},
};

export default RentalCarParkingContent;
