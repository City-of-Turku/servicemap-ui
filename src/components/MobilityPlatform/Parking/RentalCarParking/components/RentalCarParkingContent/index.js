import { withStyles } from '@mui/styles';
import { injectIntl } from 'react-intl';
import RentalCarParkingContent from './RentalCarParkingContent';
import styles from './styles';

export default withStyles(styles)(injectIntl(RentalCarParkingContent));
