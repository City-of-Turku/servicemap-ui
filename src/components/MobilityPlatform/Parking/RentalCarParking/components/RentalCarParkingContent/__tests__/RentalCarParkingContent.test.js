/* eslint-disable max-len */
// Link.react.test.js
import React from 'react';
import { getRenderWithProviders } from '../../../../../../../../jestUtils';
import finnishTranslations from '../../../../../../../i18n/fi';
import { initialState } from '../../../../../../../redux/reducers/user';
import RentalCarParkingContent from '../index';

const mockProps = {
  item: {
    name_fi: 'Parkkipaikka',
    address_fi: 'Testikatu 1',
    extra: {
      Paikkoja_y: 5,
      Max_aika_h: 1,
      Rajoit_lis: {
        fi: 'Lisäkilpi',
      },
      Saavutetta: 'Vapaa pääsy',
    },
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<RentalCarParkingContent />', () => {
  it('does match snapshot', () => {
    const { container } = renderWithProviders(<RentalCarParkingContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<RentalCarParkingContent {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(mockProps.item.name_fi);
    expect(p[1].textContent).toContain(`Osoite: ${mockProps.item.address_fi}`);
    expect(p[2].textContent).toContain(`Parkkipaikkojen määrä: ${mockProps.item.extra.Paikkoja_y}`);
    expect(p[3].textContent).toContain(`Pysäköinnin enimmäisaika: ${mockProps.item.extra.Max_aika_h}`);
    expect(p[4].textContent).toContain(mockProps.item.extra.Rajoit_lis.fi);
    expect(p[5].textContent).toContain(finnishTranslations['mobilityPlatform.content.publicParking.access']);
  });
});
