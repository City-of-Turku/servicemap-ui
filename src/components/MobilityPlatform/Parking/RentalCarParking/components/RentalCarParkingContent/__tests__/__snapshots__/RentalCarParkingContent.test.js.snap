// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RentalCarParkingContent /> does match snapshot 1`] = `
<div>
  <div
    class="injectIntl(RentalCarParkingContent)-container-1"
  >
    <div
      class="injectIntl(RentalCarParkingContent)-headerContainer-2"
    >
      <div
        class="css-qpxa1n"
      >
        <p
          class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
        >
          Parkkipaikka
        </p>
      </div>
    </div>
    <div
      class="injectIntl(RentalCarParkingContent)-textContainer-3"
    >
      <div
        class="css-qpxa1n"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Osoite: Testikatu 1
        </p>
      </div>
      <div
        class="injectIntl(RentalCarParkingContent)-margin-4"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Parkkipaikkojen määrä: 5
        </p>
      </div>
      <div
        class="injectIntl(RentalCarParkingContent)-margin-4"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Pysäköinnin enimmäisaika: 1 tuntia
        </p>
      </div>
      <div
        class="css-qpxa1n"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Lisäkilpi
        </p>
      </div>
      <div
        class="injectIntl(RentalCarParkingContent)-margin-4"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Esteetön pääsy
        </p>
      </div>
    </div>
  </div>
</div>
`;
